pkg/atf/reader.go

```diff
@@ -0,0 +1,188 @@
+// package atf or acceptance-test-framework consists of helper files to parse,
+// validate and run acceptance test with vmaas specified terraform acceptance
+// test case format
+package atf
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/spf13/viper"
+)
+
+type accConfig struct {
+	config      string
+	validations []validation
+}
+
+type validation struct {
+	isJSON bool
+	key    string
+	value  interface{}
+}
+
+type reader struct {
+	t           *testing.T
+	isResource  bool
+	name        string
+	expectError *regexp.Regexp
+}
+
+func newReader(t *testing.T, isResource bool, name string) *reader {
+	return &reader{
+		t:          t,
+		isResource: isResource,
+		name:       name,
+	}
+}
+
+func (r *reader) fatalf(format string, v ...interface{}) {
+	r.t.Fatalf("[acc-test] test case for resource "+r.name+" failed. "+format, v...)
+}
+
+func (r *reader) skipf(format string, v ...interface{}) {
+	r.t.Skipf("[acc-test] test case for resource "+r.name+" is skipped. "+format, v...)
+}
+
+func (r *reader) getViperConfig(version string) *viper.Viper {
+	tfName := getLocalName(r.name)
+	if path := os.Getenv(accTestPath); path != "" {
+		accTestPath = path
+	}
+	var postfix string
+	if version != "" {
+		postfix = fmt.Sprintf("-%s", version)
+	}
+	v := viper.New()
+	v.SetConfigFile(fmt.Sprintf("%s/%s/%s%s.yaml", accTestPath, getTag(r.isResource), tfName, postfix))
+	err := v.ReadInConfig()
+	if err != nil {
+		r.skipf("error while reading config, %v", err)
+	}
+
+	return v
+}
+
+func parseMeta(data string) string {
+	exp := `%(rand_int)(\{[0-9]+,[0-9]+\})?`
+	reg := regexp.MustCompile(exp)
+
+	matches := reg.FindAllString(data, -1)
+	var randInt int
+	r := newRand()
+	for _, m := range matches {
+		offReg := regexp.MustCompile(`[0-9]+,[0-9]`)
+		numStr := offReg.FindString(m)
+		if numStr != "" {
+			intSplit := strings.Split(numStr, ",")
+			n1 := toInt(intSplit[0])
+			n2 := toInt(intSplit[1])
+			randInt = r.Intn(n2-n1) + n1
+		} else {
+			randInt = r.Intn(randMaxLimit)
+		}
+		data = strings.Replace(data, m, strconv.Itoa(randInt), 1)
+	}
+
+	return data
+}
+
+func (r *reader) parseValidations(vip *viper.Viper, i int) []validation {
```

Comment:
Currently validation only supports `tf` and `json` only. But in the long run we may also want to validate the length of an array of size of some fields. So please think about adding those validation in near future